Sources/StreamChatUI/ChatMessageList/MessageComposer/ChatMessageComposerMentionCellView_Tests.swift

```diff
@@ -0,0 +1,160 @@
+    private static var defaultCellWidth: CGFloat = 300
+
+    private var chatUserOffline: _ChatUser<NoExtraData>!
+    private var chatUserOnline: _ChatUser<NoExtraData>!
+    private var chatUserNoName: _ChatUser<NoExtraData>!
+
+    override func setUp() {
+        super.setUp()
+        chatUserOffline = .mock(
+            id: "darkside37",
+            name: "Mr Vader",
+            imageURL: TestImages.vader.url,
+            isOnline: false
+        )
+
+        chatUserOnline = .mock(
+            id: "darkside37",
+            name: "Mr Vader",
+            imageURL: TestImages.vader.url,
+            isOnline: true
+        )
+
+        chatUserNoName = .mock(
+            id: "yoda",
+            imageURL: TestImages.yoda.url,
+            isOnline: true
+        )
+    }
+
+    func test_emptyAppearance() {
+        let view = ChatMessageComposerMentionCellView().withoutAutoresizingMaskConstraints
+        view.widthAnchor.constraint(equalToConstant: Self.defaultCellWidth).isActive = true
+
+        AssertSnapshot(view)
+    }
+
+    func test_defaultAppearance() {
+        let view = ChatMessageComposerMentionCellView().withoutAutoresizingMaskConstraints
+        view.widthAnchor.constraint(equalToConstant: Self.defaultCellWidth).isActive = true
+
+        view.content = chatUserOnline
+        AssertSnapshot(view, suffix: "online indicator visible")
+
+        view.content = chatUserOffline
+        AssertSnapshot(view)
+
+        view.content = chatUserNoName
+        AssertSnapshot(view, suffix: "user name not set")
+    }
+
+    func test_appearanceCustomization_usingUIConfig() {
+        class RectIndicator: UIView {
+            override func didMoveToSuperview() {
+                super.didMoveToSuperview()
+                backgroundColor = .green
+                widthAnchor.constraint(equalTo: heightAnchor, multiplier: 1).isActive = true
+            }
+        }
+
+        class CustomAvatarView: _ChatChannelAvatarView<NoExtraData> {
+            override func didMoveToSuperview() {
+                super.didMoveToSuperview()
+                backgroundColor = .green
+                widthAnchor.constraint(equalTo: heightAnchor, multiplier: 1).isActive = true
+            }
+        }
+
+        var config = UIConfig()
+        config.channelList.channelListItemSubviews.onlineIndicator = RectIndicator.self
+        config.messageComposer.mentionAvatarView = CustomAvatarView.self
+
+        let view = ChatMessageComposerMentionCellView().withoutAutoresizingMaskConstraints
+        view.widthAnchor.constraint(equalToConstant: Self.defaultCellWidth).isActive = true
+
+        view.uiConfig = config
+        view.content = chatUserOnline
+        AssertSnapshot(view, suffix: "online indicator visible")
+
+        view.content = chatUserOffline
+        AssertSnapshot(view)
+
+        view.content = chatUserNoName
+        AssertSnapshot(view, suffix: "user name not set")
```

Comment:
Does it make sense to test here all variants? 🤔 We already test them in the defaultAppearance. Here we are only interested in testing if the customization works,right?

Reply:
Actually we are interested how customization work in various scenarios and with different traits. Now that we finally made it with `contentSize` and Fonts working in tests I think this is quite okay.